### ISSUE TITLE
AxisArray recipes

### ISSUE THREAD

**Issue by jw3126** [JuliaPlots/Plots.jl#1705]
I have a tiny unregisterd package [AxisArrayPlots](https://github.com/jw3126/AxisArrayPlots.jl) that has recipes for [AxisArrays](https://github.com/JuliaArrays/AxisArrays.jl). The wish to have this functionality available via METADATA came up, see here https://github.com/jw3126/AxisArrayPlots.jl/issues/1.
Instead of adding the tiny package itself to METADATA I would prefer to do a PR to another package in the plots ecosystem. Any suggestions for a candidate?

**Comment by daschw** [comment]
From our perspective the ideal place for such recipes would be the package defining the type, so in this case AxisArrays.jl - if the maintainers are willing to add the RecipesBase dependency.

Otherwise, we have two places to put recipe code: StatPlots (only recipes related to statistics) and PlotRecipes (for everything else). For both we don't really want to blow up our dependencies, because this means that everyone who wants to use one of these recipes has to install all the other packages as well.

**Comment by jw3126** [comment]
Yeah, [maintainers of AxisArrays](https://github.com/JuliaArrays/AxisArrays.jl/pull/101) were not exactly keen on this, either.

**Comment by mkborregaard** [comment]
I agree actually that plotrecipes don't belong in AxisArrays given that it is so basal. It would belong in PlotRecipes, and I don't think a dep on AxisArrays is something to worry about. I do agree though with the comment on that issue that it's not intuitive how to plot them - AFAICS the only necessary behaviour of an axisarray recipe would be to provide default axis labels and ticks? This is something we've talked about in other contexts too: https://github.com/JuliaPlots/Plots.jl/issues/1072 but I'm not sure a `user recipe` is an intuitive way to provide this.

Maybe as a `type recipe`? http://docs.juliaplots.org/latest/recipes/#type-recipes_1

**Comment by jw3126** [comment]
* For 1d AxisArray axis label+ticks I fully agree, that axis label+ticks is the only necessary behaviour. And this is what the package does.
* For 2d it is not so clear. There is already a method 
```julia
xs = 1:3
ys = 1:4
data = randn(3,4)
plot(xs,ys, data)
```
that gives a contour plot. So the packages also massages an `AxisMatrix` into this.

By as a `type recipe` you mean another wrapper type around `AxisArray` should be introduced?

**Comment by mkborregaard** [comment]
No, I mean I don't agree with the logic of dispatching an AxisMatrix to `surface` necessarily. IMHO it should just behave as a normal Matrix, + possibly label/ticks.

**Comment by jw3126** [comment]
Ok default could be normal Matrix + xlabel/ticks. If seriestype is 2d e.g. heatmap, surface I would like to magically have the same behaviour as right now. Are you okay with that?
